apps/native/app/src/hoc/offline-hoc.tsx

```diff
@@ -0,0 +1,53 @@
+import { impactAsync, ImpactFeedbackStyle } from 'expo-haptics'
+import { ReactNode, useEffect } from 'react'
+import { Navigation } from 'react-native-navigation'
+import { useAuthStore } from '../stores/auth-store'
+import { useOfflineStore } from '../stores/offline-store'
+import { ComponentRegistry as CR } from '../utils/component-registry'
+
+interface OfflineHocProps {
+  children: ReactNode
+}
+
+export const OfflineHoc = ({ children }: OfflineHocProps) => {
+  const isConnected = useOfflineStore(({ isConnected }) => isConnected)
+  const bannerHasBeenShown = useOfflineStore(
+    ({ bannerHasBeenShown }) => bannerHasBeenShown,
+  )
+  const bannerVisible = useOfflineStore(({ bannerVisible }) => bannerVisible)
+  const setBannerHasBeenShown = useOfflineStore(
+    ({ setBannerHasBeenShown }) => setBannerHasBeenShown,
+  )
+
+  const lockScreenActivatedAt = useAuthStore(
+    ({ lockScreenActivatedAt }) => lockScreenActivatedAt,
```

Comment:
They have been doing it wrong. Its a recommended practice with Zustand version 3. It will cause rerenders when other unrelated properties in the store update.
However if we udate to version 4 then we can use useShallow hook. But until then this is the recommended practice although it is clearly uglier 
https://github.com/pmndrs/zustand#selecting-multiple-state-slices